python/ray/experimental/channel/common.py

```diff
@@ -533,7 +552,14 @@ def _run(self, res):
         for i, channel in enumerate(self._output_channels):
             idx = self._output_idxs[i]
             res_i = _adapt(res, idx, self._is_input)
-            channel.write(res_i)
+            while True:
+                try:
+                    channel.write(res_i, timeout=1)
+                except ray.exceptions.RayChannelTimeoutError:
+                    if sys.is_finalizing():
+                        break
+                else:
+                    break
```

Comment:
nit: can we unify with the same code pattern as in L361-L371? i.e., L561-L562 does the same thing as L365.